cal/cal/models.py

```diff
@@ -584,6 +584,40 @@ def query(self, calendar_ids=None, start=None, end=None):
         return events_qs
 
 
+    def queryWeek(self, calendar_ids=None, start=None, end=None):
+        if calendar_ids and self.calendar:
+            message = "Calendar ids can only be specified for ColorCategories without a calendar field attached."
+            raise InvalidParameterException(message)
+
+        if self.calendar:
+            calendars = [self.calendar]
+        else:
+            calendars = self.user.profile.get_calendars_for_calendarids(calendar_ids)
+
+        querysets = [
+                GEvent.objects
+                .filter(calendar__user=self.user, calendar=calendar, color_index=self.color_index)
+                .exclude(all_day_event=True)
+                for calendar in calendars
+                ]
+
+        # Union over the querysets
+        events_qs = reduce(lambda qs1, qs2: qs1 | qs2, querysets)
+
+        queryset = []
+        events = events_qs.order_by('start')
+        i = 0
+        start = events[0].start # We have to start with this outside for loop, seems sketch
+        while i < len(events):
+            end = start + timedelta(days=7)
```

Comment:
Yes. That's the purpose of timedelta(days=7), is basically we start at a start date, then iterate through until we hit an event that is not in the week's threshold. Then we'll set our end date to our start date.